src/index.js

```diff
@@ -1,32 +1,41 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/'
+const xhttp = new XMLHttpRequest()
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+const fetchDataByPromise = url => {   
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = function () {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) {
+          const response = JSON.parse(xhttp.responseText)
+          return resolve(response)
+        } else{
+          return reject(new Error('Hubo un error :('))
+        } 
+      }
     }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
-};
+    xhttp.responseType = 'json'
+    xhttp.open('GET', url, false)
+    xhttp.send()
+	})
+}
 
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
-    console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
-  });
-});
\ No newline at end of file
+const fetchData = async url => {
+  try {
+    console.log('Primer llamado...')
+    const world = await fetchDataByPromise(url)
+    console.log('Segundo llamado...')
+    const chapters = await fetchDataByPromise(`${API}${world.results[0].id}`)
+    console.log('Tercero Llamado...')
+    const characters = await fetchDataByPromise(`${chapters.origin.url}`)
```

Comment:
Si no vamos a unir mas de un elemento, podemos pasar el parametro sin los templates literals.

```suggestion
    const characters = await fetchDataByPromise(chapters.origin.url)
```